sdk/OoyalaSkinSDK/discoveryPanel.js

```diff
@@ -69,6 +81,30 @@ var DiscoveryPanel = React.createClass({
     }
   },
 
+  onStatusPressed: function() {
+    this.setState({showCircularStatus: false});
+  },
+
+  setCounterTime: function(time) {
+    this.setState({
+      currentCounterVal: time,
+      counterLimit: time,
+      showCircularStatus: true,
+    });
+  },
+
+  updateTimer: function(row) {
+    if (this.state.currentCounterVal == 0) {
+      this.onRowSelected(row);
+    } else {
+      var self = this;
+      setTimeout(function() {
```

Comment:
This is only slightly concerning to me - It is the UI causing things to happen.  I don't have the best idea of if there is a better way to do this, but I'll voice my concern at least, and keep reading